File: DS & Algo/sieve.cpp
#include <iostream>

int capacity = 1000;
bool primes[1000];

void sieve() {
	primes[1] = false;

	int limit = capacity;

	for(int i = 2; i < limit; ++i) {
		primes[i] = true;
	}

	for(int i = 2; i < limit; ++i) {
		if(primes[i]) {
			for(int j = 2; j*i < limit; ++j) {
				primes[i*j] = false;
			}
		}
	}
}

bool isPrime(int input) {
	return primes[input];
}

int main() {

	sieve();

	std::cout << isPrime(3) << std::endl;
	std::cout << isPrime(8) << std::endl;
	std::cout << isPrime(13) << std::endl;
	std::cout << isPrime(69) << std::endl;

	return 0;
}

File: DS & Algo/Sieve.java
import java.util.Arrays;

class Sieve {

	static int capacity = 1_000_000;
	static boolean[] primes = new boolean[capacity];

	public static void main(String args[]) {
		fillSieve();

		// few test cases
		System.out.println(isPrime(0));
		System.out.println(isPrime(1));
		System.out.println(isPrime(2));
		System.out.println(isPrime(16));
		System.out.println(isPrime(101));
	}
	public static void fillSieve() {

		// everything is a prime at first
		Arrays.fill(primes, true);

		// 0 and 1 are not prime
		primes[0] = primes[1] = false;

		for(int i = 2; i < primes.length; i++) {

			// if this is a prime number
			if(primes[i]) {

				// all of the primes numbers multiples are not prime
				for(int j = 2; i*j < primes.length; j++)
					primes[i*j] = false;
			}
		}
	}

	public static boolean isPrime(int n) {
		return primes[n];
	}
}
